python/adios2/io.py

```diff
@@ -67,13 +67,21 @@ def define_attribute(
                 Not used if variable_name is empty
         """
 
-        # string or list of strings passed on as is
-        if isinstance(content, list) and len(content) > 0 and isinstance(content[0], str):
+        if isinstance(content, (np.ndarray, str)):
             return Attribute(self.impl, name, content, variable_name, separator)
-        if isinstance(content, str):
+
+        if isinstance(content, list):
+            if len(content) > 0 and isinstance(content[0], str):
+                # string list passed on, becomes std::vector<std::string>
+                return Attribute(self.impl, name, content, variable_name, separator)
+            # non string lists needs to be converted to numpy array
+            content = np.array(content)
+            return Attribute(self.impl, name, content, variable_name, separator)
```

Comment:
Why do we need to convert attributes list (no string list) to numpy arrays?

Reply:
Because we had no glue functions. Now I added them for lists (vectors) and values (pybind11::object).